src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -110,7 +108,8 @@ public static unsafe string PtrToStringUTF8(IntPtr ptr)
             return string.CreateStringFromEncoding((byte*)ptr, nbBytes, Encoding.UTF8);
         }
 
-        public static unsafe string PtrToStringUTF8(IntPtr ptr, int byteLen)
+        // TODO-NULLABLE: This has different behavior from the other PtrToString(IntPtr, int) functions
```

Comment:
The other APIs return `null` for `ptr == IntPtr.Zero` on the overload that doesn't take a length; and throw for the input that does take a length.

We should probably make these consistent.

Reply:
CC. @jkoritzinsky.

Reply:
> We should probably make these consistent.

Can you open a separate issue for that?

Reply:
Logged https://github.com/dotnet/coreclr/issues/23723